### PR TITLE
Merging 2.1 to 2.2

### PR DESCRIPTION
**Pull request by JunTaoLuo**
Rest of the changes in https://github.com/aspnet/AspNetCore/pull/6569

### CODE REVIEW

**Review by ajaybhargavb — COMMENTED** [review]
Razor changes LGTM :shipit:

File: eng/Baseline.Designer.props [hunk 1]
```diff
@@ -1004,9 +1004,9 @@
     <BaselinePackageReference Include="Microsoft.Extensions.Logging.Abstractions" Version="[2.2.0, )" />
     <BaselinePackageReference Include="Microsoft.Extensions.Options" Version="[2.2.0, )" />
     <BaselinePackageReference Include="Microsoft.Net.Http.Headers" Version="[2.2.0, )" />
-    <BaselinePackageReference Include="System.Memory" Version="[4.5.1, )" />
+    <BaselinePackageReference Include="System.Memory" Version="[4.5.2, )" />
```
Comment (natemcmaster) [inline eng/Baseline.Designer.props]: Changes to this file shouldn't be merged.

**Review by natemcmaster — APPROVED** [review]
One little thing to revert, otherwise looks good. Bulk of the changes should be signed off by @pakrym and @Tratcher

**Review by Tratcher — APPROVED** [review]
For Google Auth

File: eng/Baseline.Designer.props (hunk 1 above)
Comment (JunTaoLuo) [reply eng/Baseline.Designer.props]: D'oh. I caught the other similar changes but I missed one it seems.

**Review by pakrym — APPROVED** [review]
SiteExtension LGTM

**Review by mikaelm12 — APPROVED** [review]
With respect to 2.1.8 changes

File: build/SharedFx.targets [hunk 2]
```diff
@@ -43,6 +43,28 @@
 
   <Target Name="_BuildSharedFxProjects" DependsOnTargets="GeneratePropsFiles;ResolveCommitHash">
 
+  <Target Name="BuildSiteExtension" DependsOnTargets="BuildMetapackages" Condition="'$(BuildSiteExtension)' == 'true'" >
```
Comment (JunTaoLuo) [inline build/SharedFx.targets]: Oops this merged poorly.

**Pull request merged**